`tapi` link failure due to duplicate or undefined symbols

Occurs when building latest v1300 HEAD on Ubuntu 20.04 & 24.04.

Log:
```
[1980/1981] Linking CXX executable bin/tapi
FAILED: bin/tapi
: && /usr/bin/c++  -I/home/test/apple-libtapi/src/llvm/projects/clang/include/ -I/home/test/apple-libtapi/build/projects/clang/include/ -fPIC -fvisibility-inlines-hidden -Werror=date-time -w -fdiagnostics-color -ffunction-sections -fdata-sections -fno-common -Woverloaded-virtual -fno-strict-aliasing -O3 -DNDEBUG  -Wl,-rpath-link,/home/test/apple-libtapi/build/./lib  -Wl,-O3 -Wl,--gc-sections  -Wl,--version-script,"/home/test/apple-libtapi/build/tools/clang/tools/tapi/tools/tapi/tapi.exports" tools/clang/tools/tapi/tools/tapi/CMakeFiles/tapi.dir/tapi.cpp.o  -o bin/tapi  -Wl,-rpath,"\$ORIGIN/../lib"  -lpthread  lib/libtapiDriver.a  lib/libtapiObjCMetadata.a  lib/libtapiConfig.a  lib/libtapiFrontend.a  lib/libtapiDiagnostics.a  lib/libtapiCore.a  lib/libclangFrontend.a  lib/libclangDriver.a  lib/libclangParse.a  lib/libclangSerialization.a  lib/libclangSema.a  lib/libclangAPINotes.a  lib/libclangEdit.a  lib/libclangAnalysis.a  lib/libclangASTMatchers.a  lib/libclangAST.a  lib/libLLVMFrontendOpenMP.a  lib/libLLVMTransformUtils.a  lib/libLLVMAnalysis.a  lib/libclangToolingCore.a  lib/libclangRewrite.a  lib/libclangLex.a  lib/libclangBasic.a  lib/libLLVMOption.a  lib/libLLVMProfileData.a  lib/libtapiSDKDB.a  lib/libLLVMObject.a  lib/libLLVMBitReader.a  lib/libLLVMCore.a  lib/libLLVMRemarks.a  lib/libLLVMBitstreamReader.a  lib/libLLVMMCParser.a  lib/libLLVMMC.a  lib/libLLVMDebugInfoCodeView.a  lib/libLLVMDebugInfoMSF.a  lib/libLLVMTextAPI.a  lib/libLLVMBinaryFormat.a  lib/libLLVMSupport.a  -lrt  -ldl  -lpthread  -lm  lib/libLLVMDemangle.a && :
/usr/bin/ld: lib/libLLVMTextAPI.a(TextStubCommon.cpp.o): in function `llvm::yaml::ScalarTraits<FlowStringRef, void>::output(FlowStringRef const&, void*, llvm::raw_ostream&)':
TextStubCommon.cpp:(.text._ZN4llvm4yaml12ScalarTraitsI13FlowStringRefvE6outputERKS2_PvRNS_11raw_ostreamE+0x0): multiple definition of `llvm::yaml::ScalarTraits<FlowStringRef, void>::output(FlowStringRef const&, void*, llvm::raw_ostream&)'; lib/libtapiCore.a(TextStubCommon.cpp.o):TextStubCommon.cpp:(.text._ZN4llvm4yaml12ScalarTraitsI13FlowStringRefvE6outputERKS2_PvRNS_11raw_ostreamE+0x0): first defined here
/usr/bin/ld: lib/libLLVMTextAPI.a(TextStubCommon.cpp.o): in function `llvm::yaml::ScalarTraits<FlowStringRef, void>::input(llvm::StringRef, void*, FlowStringRef&)':
TextStubCommon.cpp:(.text._ZN4llvm4yaml12ScalarTraitsI13FlowStringRefvE5inputENS_9StringRefEPvRS2_+0x0): multiple definition of `llvm::yaml::ScalarTraits<FlowStringRef, void>::input(llvm::StringRef, void*, FlowStringRef&)'; lib/libtapiCore.a(TextStubCommon.cpp.o):TextStubCommon.cpp:(.text._ZN4llvm4yaml12ScalarTraitsI13FlowStringRefvE5inputENS_9StringRefEPvRS2_+0x0): first defined here
/usr/bin/ld: lib/libLLVMTextAPI.a(TextStubCommon.cpp.o): in function `llvm::yaml::ScalarTraits<FlowStringRef, void>::mustQuote(llvm::StringRef)':
TextStubCommon.cpp:(.text._ZN4llvm4yaml12ScalarTraitsI13FlowStringRefvE9mustQuoteENS_9StringRefE+0x0): multiple definition of `llvm::yaml::ScalarTraits<FlowStringRef, void>::mustQuote(llvm::StringRef)'; lib/libtapiCore.a(TextStubCommon.cpp.o):TextStubCommon.cpp:(.text._ZN4llvm4yaml12ScalarTraitsI13FlowStringRefvE9mustQuoteENS_9StringRefE+0x0): first defined here
/usr/bin/ld: lib/libLLVMTextAPI.a(TextStubCommon.cpp.o): in function `llvm::yaml::ScalarBitSetTraits<llvm::MachO::ArchitectureSet, void>::bitset(llvm::yaml::IO&, llvm::MachO::ArchitectureSet&)':
TextStubCommon.cpp:(.text._ZN4llvm4yaml18ScalarBitSetTraitsINS_5MachO15ArchitectureSetEvE6bitsetERNS0_2IOERS3_+0x0): multiple definition of `llvm::yaml::ScalarBitSetTraits<llvm::MachO::ArchitectureSet, void>::bitset(llvm::yaml::IO&, llvm::MachO::ArchitectureSet&)'; lib/libtapiCore.a(TextStubCommon.cpp.o):TextStubCommon.cpp:(.text._ZN4llvm4yaml18ScalarBitSetTraitsINS_5MachO15ArchitectureSetEvE6bitsetERNS0_2IOERS3_+0x0): first defined here
/usr/bin/ld: lib/libLLVMTextAPI.a(TextStubCommon.cpp.o): in function `llvm::yaml::ScalarTraits<llvm::MachO::Architecture, void>::output(llvm::MachO::Architecture const&, void*, llvm::raw_ostream&)':
TextStubCommon.cpp:(.text._ZN4llvm4yaml12ScalarTraitsINS_5MachO12ArchitectureEvE6outputERKS3_PvRNS_11raw_ostreamE+0x0): multiple definition of `llvm::yaml::ScalarTraits<llvm::MachO::Architecture, void>::output(llvm::MachO::Architecture const&, void*, llvm::raw_ostream&)'; lib/libtapiCore.a(TextStubCommon.cpp.o):TextStubCommon.cpp:(.text._ZN4llvm4yaml12ScalarTraitsINS_5MachO12ArchitectureEvE6outputERKS3_PvRNS_11raw_ostreamE+0x0): first defined here
/usr/bin/ld: lib/libLLVMTextAPI.a(TextStubCommon.cpp.o): in function `llvm::yaml::ScalarTraits<llvm::MachO::Architecture, void>::input(llvm::StringRef, void*, llvm::MachO::Architecture&)':
TextStubCommon.cpp:(.text._ZN4llvm4yaml12ScalarTraitsINS_5MachO12ArchitectureEvE5inputENS_9StringRefEPvRS3_+0x0): multiple definition of `llvm::yaml::ScalarTraits<llvm::MachO::Architecture, void>::input(llvm::StringRef, void*, llvm::MachO::Architecture&)'; lib/libtapiCore.a(TextStubCommon.cpp.o):TextStubCommon.cpp:(.text._ZN4llvm4yaml12ScalarTraitsINS_5MachO12ArchitectureEvE5inputENS_9StringRefEPvRS3_+0x0): first defined here
/usr/bin/ld: lib/libLLVMTextAPI.a(TextStubCommon.cpp.o): in function `llvm::yaml::ScalarTraits<llvm::MachO::Architecture, void>::mustQuote(llvm::StringRef)':
TextStubCommon.cpp:(.text._ZN4llvm4yaml12ScalarTraitsINS_5MachO12ArchitectureEvE9mustQuoteENS_9StringRefE+0x0): multiple definition of `llvm::yaml::ScalarTraits<llvm::MachO::Architecture, void>::mustQuote(llvm::StringRef)'; lib/libtapiCore.a(TextStubCommon.cpp.o):TextStubCommon.cpp:(.text._ZN4llvm4yaml12ScalarTraitsINS_5MachO12ArchitectureEvE9mustQuoteENS_9StringRefE+0x0): first defined here
/usr/bin/ld: lib/libLLVMTextAPI.a(TextStubCommon.cpp.o): in function `llvm::yaml::ScalarTraits<SwiftVersion, void>::output(SwiftVersion const&, void*, llvm::raw_ostream&)':
TextStubCommon.cpp:(.text._ZN4llvm4yaml12ScalarTraitsI12SwiftVersionvE6outputERKS2_PvRNS_11raw_ostreamE+0x0): multiple definition of `llvm::yaml::ScalarTraits<SwiftVersion, void>::output(SwiftVersion const&, void*, llvm::raw_ostream&)'; lib/libtapiCore.a(TextStubCommon.cpp.o):TextStubCommon.cpp:(.text._ZN4llvm4yaml12ScalarTraitsI12SwiftVersionvE6outputERKS2_PvRNS_11raw_ostreamE+0x0): first defined here
/usr/bin/ld: lib/libLLVMTextAPI.a(TextStubCommon.cpp.o): in function `llvm::yaml::ScalarTraits<SwiftVersion, void>::input(llvm::StringRef, void*, SwiftVersion&)':
TextStubCommon.cpp:(.text._ZN4llvm4yaml12ScalarTraitsI12SwiftVersionvE5inputENS_9StringRefEPvRS2_+0x0): multiple definition of `llvm::yaml::ScalarTraits<SwiftVersion, void>::input(llvm::StringRef, void*, SwiftVersion&)'; lib/libtapiCore.a(TextStubCommon.cpp.o):TextStubCommon.cpp:(.text._ZN4llvm4yaml12ScalarTraitsI12SwiftVersionvE5inputENS_9StringRefEPvRS2_+0x0): first defined here
/usr/bin/ld: lib/libLLVMTextAPI.a(TextStubCommon.cpp.o): in function `llvm::yaml::ScalarTraits<SwiftVersion, void>::mustQuote(llvm::StringRef)':
TextStubCommon.cpp:(.text._ZN4llvm4yaml12ScalarTraitsI12SwiftVersionvE9mustQuoteENS_9StringRefE+0x0): multiple definition of `llvm::yaml::ScalarTraits<SwiftVersion, void>::mustQuote(llvm::StringRef)'; lib/libtapiCore.a(TextStubCommon.cpp.o):TextStubCommon.cpp:(.text._ZN4llvm4yaml12ScalarTraitsI12SwiftVersionvE9mustQuoteENS_9StringRefE+0x0): first defined here
/usr/bin/ld: lib/libLLVMTextAPI.a(TextStubCommon.cpp.o): in function `llvm::yaml::ScalarTraits<std::pair<llvm::MachO::Target, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, void>::output(std::pair<llvm::MachO::Target, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > const&, void*, llvm::raw_ostream&)':
TextStubCommon.cpp:(.text._ZN4llvm4yaml12ScalarTraitsISt4pairINS_5MachO6TargetENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEEvE6outputERKSB_PvRNS_11raw_ostreamE+0x0): multiple definition of `llvm::yaml::ScalarTraits<std::pair<llvm::MachO::Target, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, void>::output(std::pair<llvm::MachO::Target, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > const&, void*, llvm::raw_ostream&)'; lib/libtapiCore.a(TextStubCommon.cpp.o):TextStubCommon.cpp:(.text._ZN4llvm4yaml12ScalarTraitsISt4pairINS_5MachO6TargetENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEEvE6outputERKSB_PvRNS_11raw_ostreamE+0x0): first defined here
/usr/bin/ld: lib/libLLVMTextAPI.a(TextStubCommon.cpp.o): in function `llvm::yaml::ScalarTraits<std::pair<llvm::MachO::Target, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, void>::input(llvm::StringRef, void*, std::pair<llvm::MachO::Target, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >&)':
TextStubCommon.cpp:(.text._ZN4llvm4yaml12ScalarTraitsISt4pairINS_5MachO6TargetENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEEvE5inputENS_9StringRefEPvRSB_+0x0): multiple definition of `llvm::yaml::ScalarTraits<std::pair<llvm::MachO::Target, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, void>::input(llvm::StringRef, void*, std::pair<llvm::MachO::Target, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >&)'; lib/libtapiCore.a(TextStubCommon.cpp.o):TextStubCommon.cpp:(.text._ZN4llvm4yaml12ScalarTraitsISt4pairINS_5MachO6TargetENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEEvE5inputENS_9StringRefEPvRSB_+0x0): first defined here
/usr/bin/ld: lib/libLLVMTextAPI.a(TextStubCommon.cpp.o): in function `llvm::yaml::ScalarTraits<std::pair<llvm::MachO::Target, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, void>::mustQuote(llvm::StringRef)':
TextStubCommon.cpp:(.text._ZN4llvm4yaml12ScalarTraitsISt4pairINS_5MachO6TargetENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEEvE9mustQuoteENS_9StringRefE+0x0): multiple definition of `llvm::yaml::ScalarTraits<std::pair<llvm::MachO::Target, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, void>::mustQuote(llvm::StringRef)'; lib/libtapiCore.a(TextStubCommon.cpp.o):TextStubCommon.cpp:(.text._ZN4llvm4yaml12ScalarTraitsISt4pairINS_5MachO6TargetENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEEvE9mustQuoteENS_9StringRefE+0x0): first defined here
/usr/bin/ld: lib/libtapiCore.a(MachOReader.cpp.o): in function `tapi::internal::readObjectiveCMetadata(llvm::object::MachOObjectFile*, tapi::internal::API&)':
MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0x51): undefined reference to `llvm::MachOMetadata::MachOMetadata(llvm::object::MachOObjectFile*, llvm::Error&)'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0x78): undefined reference to `vtable for llvm::MachOMetadata'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0x1c6): undefined reference to `llvm::MachOMetadata::classes() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0x264): undefined reference to `llvm::ObjCClassRef::getObjCClass() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0x281): undefined reference to `llvm::ObjCClass::getSuperClassName() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0x29e): undefined reference to `llvm::ObjCClass::getName() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0x38d): undefined reference to `llvm::ObjCClass::properties() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0x3e7): undefined reference to `llvm::ObjCProperty::getAttribute() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0x421): undefined reference to `llvm::ObjCProperty::getSetter[abi:cxx11]() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0x442): undefined reference to `llvm::ObjCProperty::getGetter[abi:cxx11]() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0x557): undefined reference to `llvm::ObjCProperty::getName() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0x5c9): undefined reference to `llvm::ObjCClass::classMethods() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0x6d1): undefined reference to `llvm::ObjCMethod::getName() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0x7a1): undefined reference to `llvm::ObjCClass::instanceMethods() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0x8a3): undefined reference to `llvm::ObjCMethod::getName() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0x959): undefined reference to `llvm::MachOMetadata::categories() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0xa90): undefined reference to `llvm::ObjCCategoryRef::getObjCCategory() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0xaad): undefined reference to `llvm::ObjCCategory::getName() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0xaca): undefined reference to `llvm::ObjCCategory::getBaseClassName() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0xbbb): undefined reference to `llvm::ObjCCategory::properties() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0xc0d): undefined reference to `llvm::ObjCProperty::getAttribute() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0xc46): undefined reference to `llvm::ObjCProperty::getSetter[abi:cxx11]() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0xc67): undefined reference to `llvm::ObjCProperty::getGetter[abi:cxx11]() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0xd8f): undefined reference to `llvm::ObjCProperty::getName() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0xecf): undefined reference to `llvm::ObjCCategory::classMethods() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0xfd1): undefined reference to `llvm::ObjCMethod::getName() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0x101f): undefined reference to `llvm::ObjCCategory::instanceMethods() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0x1053): undefined reference to `llvm::ObjCMethod::getName() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0x11f1): undefined reference to `llvm::MachOMetadata::protocols() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0x130a): undefined reference to `llvm::ObjCProtocolRef::getObjCProtocol() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0x1325): undefined reference to `llvm::ObjCProtocol::getName() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0x1401): undefined reference to `llvm::ObjCProtocol::properties() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0x1444): undefined reference to `llvm::ObjCProperty::getName() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0x146a): undefined reference to `llvm::ObjCProperty::getAttribute() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0x14a2): undefined reference to `llvm::ObjCProperty::getSetter[abi:cxx11]() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0x14c3): undefined reference to `llvm::ObjCProperty::getGetter[abi:cxx11]() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0x15f4): undefined reference to `llvm::ObjCProtocol::classMethods() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0x1635): undefined reference to `llvm::ObjCMethod::getName() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0x1705): undefined reference to `llvm::ObjCProtocol::optionalClassMethods() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0x1759): undefined reference to `llvm::ObjCMethod::getName() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0x1829): undefined reference to `llvm::ObjCProtocol::instanceMethods() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0x1871): undefined reference to `llvm::ObjCMethod::getName() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0x1943): undefined reference to `llvm::ObjCProtocol::optionalInstanceMethods() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0x199e): undefined reference to `llvm::ObjCMethod::getName() const'
/usr/bin/ld: MachOReader.cpp:(.text._ZN4tapi8internalL22readObjectiveCMetadataEPN4llvm6object15MachOObjectFileERNS0_3APIE+0x1ca5): undefined reference to `llvm::MachOMetadata::getAllPotentiallyDefinedSelectors(llvm::StringSet<llvm::MallocAllocator>&) const'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

Configure command:
```
cmake -B build -S src/llvm -G Ninja \
	-DLLVM_TARGETS_TO_BUILD="X86" \
	-DLLVM_INCLUDE_TESTS=OFF \
	-DCLANG_INCLUDE_TESTS=OFF \
	-DTAPI_INCLUDE_TESTS=OFF \
	-DLLVM_ENABLE_WARNINGS=OFF \
	-DLLVM_ENABLE_PROJECTS="clang;tapi" \
	-DCMAKE_CXX_FLAGS="-I${PWD}/src/llvm/projects/clang/include/ -I${PWD}/build/projects/clang/include/" \
	-DCMAKE_BUILD_TYPE=Release 
```

--- a/src/tapi/lib/Core/CMakeLists.txt
+++ b/src/tapi/lib/Core/CMakeLists.txt
@@ -35,4 +35,6 @@ add_tapi_library(tapiCore
 
   LINK_LIBS
   clangBasic
+  PRIVATE
+    tapiObjCMetadata
   )

--- a/src/tapi/lib/Core/TextStubCommon.cpp
+++ b/src/tapi/lib/Core/TextStubCommon.cpp
@@ -20,15 +20,15 @@ namespace llvm {
 namespace yaml {
 
 using Impl = ScalarTraits<StringRef>;
-void ScalarTraits<FlowStringRef>::output(const FlowStringRef &value, void *ctx,
+inline void ScalarTraits<FlowStringRef>::output(const FlowStringRef &value, void *ctx,
                                          raw_ostream &os) {
   Impl::output(value, ctx, os);
 }
-StringRef ScalarTraits<FlowStringRef>::input(StringRef value, void *ctx,
+inline StringRef ScalarTraits<FlowStringRef>::input(StringRef value, void *ctx,
                                              FlowStringRef &out) {
   return Impl::input(value, ctx, out.value);
 }
-QuotingType ScalarTraits<FlowStringRef>::mustQuote(StringRef name) {
+inline QuotingType ScalarTraits<FlowStringRef>::mustQuote(StringRef name) {
   return Impl::mustQuote(name);
 }
 
@@ -57,7 +57,7 @@ void ScalarEnumerationTraits<PlatformKind>::enumeration(
   io.enumCase(platform, "bridgeos", PlatformKind::bridgeOS);
 }
 
-void ScalarBitSetTraits<ArchitectureSet>::bitset(IO &io,
+inline void ScalarBitSetTraits<ArchitectureSet>::bitset(IO &io,
                                                  ArchitectureSet &archs) {
 #define ARCHINFO(arch, type, subtype, numbits)                                 \
   io.bitSetCase(archs, #arch, 1U << static_cast<int>(AK_##arch));
@@ -65,16 +65,16 @@ void ScalarBitSetTraits<ArchitectureSet>::bitset(IO &io,
 #undef ARCHINFO
 }
 
-void ScalarTraits<Architecture>::output(const Architecture &value, void *,
+inline void ScalarTraits<Architecture>::output(const Architecture &value, void *,
                                         raw_ostream &os) {
   os << value;
 }
-StringRef ScalarTraits<Architecture>::input(StringRef scalar, void *,
+inline StringRef ScalarTraits<Architecture>::input(StringRef scalar, void *,
                                             Architecture &value) {
   value = getArchitectureFromName(scalar);
   return {};
 }
-QuotingType ScalarTraits<Architecture>::mustQuote(StringRef) {
+inline QuotingType ScalarTraits<Architecture>::mustQuote(StringRef) {
   return QuotingType::None;
 }
 
@@ -93,7 +93,7 @@ QuotingType ScalarTraits<PackedVersion>::mustQuote(StringRef) {
   return QuotingType::None;
 }
 
-void ScalarTraits<SwiftVersion>::output(const SwiftVersion &value, void *,
+inline void ScalarTraits<SwiftVersion>::output(const SwiftVersion &value, void *,
                                         raw_ostream &os) {
   switch (value) {
   case 1:
@@ -113,7 +113,7 @@ void ScalarTraits<SwiftVersion>::output(const SwiftVersion &value, void *,
     break;
   }
 }
-StringRef ScalarTraits<SwiftVersion>::input(StringRef scalar, void *,
+inline StringRef ScalarTraits<SwiftVersion>::input(StringRef scalar, void *,
                                             SwiftVersion &value) {
   value = StringSwitch<SwiftVersion>(scalar)
               .Case("1.0", 1)
@@ -129,7 +129,7 @@ StringRef ScalarTraits<SwiftVersion>::input(StringRef scalar, void *,
 
   return StringRef();
 }
-QuotingType ScalarTraits<SwiftVersion>::mustQuote(StringRef) {
+inline QuotingType ScalarTraits<SwiftVersion>::mustQuote(StringRef) {
   return QuotingType::None;
 }
 
@@ -171,7 +171,7 @@ QuotingType ScalarTraits<AvailabilityInfo>::mustQuote(StringRef) {
   return QuotingType::None;
 }
 
-void ScalarTraits<UUID>::output(const UUID &value, void *c, raw_ostream &os) {
+inline void ScalarTraits<UUID>::output(const UUID &value, void *c, raw_ostream &os) {
   auto *ctx = reinterpret_cast<YAMLContext *>(c);
   assert(ctx);
 
@@ -180,7 +180,7 @@ void ScalarTraits<UUID>::output(const UUID &value, void *c, raw_ostream &os) {
   else
     os << value.first << ": " << value.second;
 }
-StringRef ScalarTraits<UUID>::input(StringRef scalar, void *c, UUID &value) {
+inline StringRef ScalarTraits<UUID>::input(StringRef scalar, void *c, UUID &value) {
   auto split = scalar.split(':');
   auto arch = split.first.trim();
   auto uuid = split.second.trim();
@@ -191,7 +191,7 @@ StringRef ScalarTraits<UUID>::input(StringRef scalar, void *c, UUID &value) {
   value.second = uuid.str();
   return {};
 }
-QuotingType ScalarTraits<UUID>::mustQuote(StringRef) {
+inline QuotingType ScalarTraits<UUID>::mustQuote(StringRef) {
   return QuotingType::Single;
 }
 

